### DISCUSSION TITLE
How to embed a netlist inside a netlist?

### DISCUSSION THREAD

**Discussion by Kreijstal** [mph-/lcapy#126] [Q&A]
I want to play `puzzle`, that means I want to take what I already programmed, and simply append something to its ports, it can be either be as a "box" or as a tiny version of the circuit let me give you an example.
 Say for example I create a low pass filter, this circuit has no voltage sources, or current sources. After plotting or getting the transfer function for this circuit,  I want to "append" the voltage or maybe stack the same low pass filter again, how can you do this?
 
 If this question is not clear, let me know and I will give better examples of what I mean.

**Comment by mph-** [comment]
You can add new components to a netlist but there is no built-in method to join two netlists.   I'll have a think about this.

The alternative is to take your transfer function and multiply it by the Laplace or Fourier transform of the input signal to find the transform of the output signal.

**Reply by Kreijstal** [reply]
This doesn't work in the general case, because of the feedback loops, unless you separate them with opamps.

**Reply by Kreijstal** [reply]
But it would still be nice to have defined components that you could simply use maybe also parameterized

**Reply by mph-** [reply]
I think you need to explain what you would like to do in more detail.  Yes, you can isolate circuits with an opamp or VCVS but I thought you just wanted to connect different netlists together.

**Reply by Kreijstal** [reply]
Yes exactly, like appending multiple two port networks together, or reusing filters you've already written

**Reply by Kreijstal** [reply]
Something like substck or spice subcircuits https://youspice.com/creating-a-spice-subcircuit-subckt-manually/

**Reply by mph-** [reply]
Yeah, that's on the to do list.  I'm currently busy with the GUI version of Lcapy.
